### ISSUE TITLE
Filament sensor reporting no filament

### ISSUE THREAD

**Issue by SG1EmberWolf** [pfn/EnragedRabbitOnRRF#3]
I can't get the filament sensor to report any filament.  I've changed it to the appropriate pin, and looking at reprap documentation, it looks like sensing is disabled unless printing from SD card.  How do I go about setting it up just to get it properly working with sending filament from the MMU to the toolhead?  When I try to run M591 D0, I get no data received.  The sensor itself is powered and the eyes are flashing.  I've verified with a multimeter that the sense pin is going high and low.

**Comment by pfn** [comment]
Which pin is the sensor plugged into? Make sure that pin is correctly selected, of course. I have 4 of these sensors across 4 printers running RRF on different boards: duet2wifi, duet3mb6hc, big dipper, and octopus pro 429. They all function as expected with the provided configuration, other than picking the correct pins.

You can further verify if the sensor is configured using the correct pin by switching to pulse-counting mode, and verifying that the pulse counting variable indeed counts up.

`M98 P"ercf/lib/pulse-counting.g"` -- run filament through the sensor, observe the LEDs flicker, observe the variable `global.ercf_pulse_count` is non-zero. If this doesn't happen, then the sensor isn't configured using the right pin.

**Comment by SG1EmberWolf** [comment]
managed to solve it and was the stupidest thing ever.  Pin and code were all correct.  Sensor was too close to the BMG gear.  When I was calibrating it before assembly, I was getting reliable flashing, but after assembly, I finally managed to notice a reliable "drag" feeling when running filament through it.  Sensor was too close to get readings and best guess is the little black tabs between the emitter and collector ended up mashing with the gear to block any reading.  Fixed now and working as expected.  My sensor is connected to a duet toolboard pin since my extruder is connected to it and any filament monitor has to be on the same board as the extruder.
